File: LSP+ISP/Solution/MarsRoverKata/AlienBehavior.cs
using MarsRoverKata.Commands;
using System;

namespace MarsRoverKata
{
    public class AlienBehavior
    {
        private readonly Random _random;

        public AlienBehavior()
        {
            _random = new Random();
        }

        public void MoveAlien(Alien alien)
        {
            //switch (_random.Next(1, 4))
            //{
            //    case 1:
            //        new ForwardCommand(alien).Execute();
            //        break;
            //    case 2:
            //        new BackwardCommand(alien).Execute();
            //        break;
            //    case 3:
            //        new TurnLeftCommand(alien).Execute();
            //        MoveAlien(alien);
            //        break;
            //    case 4:
            //        new TurnRightCommand(alien).Execute();
            //        MoveAlien(alien);
            //        break;
            //}
        }
    }
}

File: LSP+ISP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/AlienBehavior.java
package MarsRoverKata;

import MarsRoverKata.Commands.BackwardCommand;
import MarsRoverKata.Commands.ForwardCommand;
import MarsRoverKata.Commands.TurnLeftCommand;
import MarsRoverKata.Commands.TurnRightCommand;

import java.util.Random;

public class AlienBehavior {
    private final Random random;

    public AlienBehavior() {
        this.random = new Random();
    }

    public void moveAlien(Alien alien) {
        switch (random.nextInt(4)) {
            case 0:
                new ForwardCommand(alien).execute();
                break;
            case 1:
                new BackwardCommand(alien).execute();
                break;
            case 2:
                new TurnLeftCommand(alien).execute();
                moveAlien(alien);
                break;
            case 3:
                new TurnRightCommand(alien).execute();
                moveAlien(alien);
                break;
        }
    }
}

